### PR TITLE
osemgrep interactive: fix the concurrency issues

### PR DESCRIPTION
**Pull request by aryx**
test plan:
in semgrep repo:
osemgrep interactive -l ocaml .
> foo

starting to type foo quickly does not generate anymore some
Lazy.Undefined exn, because access to xtarget are now protected by a lock

PR checklist:

- [x] Purpose of the code is [evident to future readers](https://semgrep.dev/docs/contributing/contributing-code/#explaining-code)
- [x] Tests included or PR comment includes a reproducible test plan
- [x] Documentation is up-to-date
- [x] A changelog entry was [added to changelog.d](https://semgrep.dev/docs/contributing/contributing-code/#adding-a-changelog-entry) for any user-facing change
- [x] Change has no security implications (otherwise, ping security team)

If you're unsure about any of this, please see:

- [Contribution guidelines](https://semgrep.dev/docs/contributing/contributing-code)!
- [One of the more specific guides located here](https://semgrep.dev/docs/contributing/contributing/)

### CODE REVIEW

File: libs/commons/Concurrency.mli [hunk 1]
```diff
@@ -0,0 +1 @@
+val with_lock : (unit -> 'a) -> Mutex.t -> 'a
```
Comment (brandonspark) [inline libs/commons/Concurrency.mli]: im amazed this doesn't already exist in our library somewhere

Comment (aryx) [reply libs/commons/Concurrency.mli]: Maybe it does. But Common.with_open_xxx didn't exist so I expected this one neither.

**Comment by aryx** [comment]
"caml_thread_initialize not implemented", new one @tpetr :)

**Comment by aryx** [comment]
the js tests are now passing @tpetr

**Pull request merged**